// @dojoengine/core@1.8.0/dist/cli/compile-abi.js
#!/usr/bin/env node

// src/cli/compile-abi.ts
import {
  readFileSync,
  writeFileSync,
  readdirSync,
  existsSync,
  mkdirSync
} from "fs";
import { join, isAbsolute, dirname, resolve } from "path";
import { fileURLToPath } from "url";
function ensureDirectory(path) {
  const directory = dirname(path);
  mkdirSync(directory, { recursive: true });
}
function resolveOutputPaths(outputOption) {
  const jsonPath = outputOption ? isAbsolute(outputOption) ? outputOption : join(process.cwd(), outputOption) : join(process.cwd(), "compiled-abi.json");
  const tsPath = jsonPath.endsWith(".json") ? `${jsonPath.slice(0, -5)}.ts` : `${jsonPath}.ts`;
  return { json: jsonPath, ts: tsPath };
}
function generateAbiTypes(paths) {
  const { json: inputPath, ts: outputPath } = paths;
  try {
    const abiContent = readFileSync(inputPath, "utf-8");
    const abiJson = JSON.parse(abiContent);
    const tsContent = `// This file is auto-generated from compiled-abi.json
// Do not edit manually

export const compiledAbi = ${JSON.stringify(abiJson, null, 2)} as const;

export type CompiledAbi = typeof compiledAbi;
`;
    ensureDirectory(outputPath);
    writeFileSync(outputPath, tsContent);
    console.log(`\u2705 Generated TypeScript types!`);
    console.log(`\u{1F4C4} Type output written to: ${outputPath}`);
    console.log(`
Usage in your code:`);
    console.log(`
import { compiledAbi } from './compiled-abi';`);
    console.log(`import { ExtractAbiTypes } from '@dojoengine/core';`);
    console.log(`
type MyAbi = ExtractAbiTypes<typeof compiledAbi>;`);
    console.log(
      `type Position = MyAbi["structs"]["dojo_starter::models::Position"];`
    );
  } catch (error) {
    console.error(`Error generating types: ${error}`);
    process.exit(1);
  }
}
function walkJsonFiles(root, entries = []) {
  const collected = [];
  for (const entry of entries) {
    const fullPath = join(root, entry.name);
    if (entry.isDirectory()) {
      const childEntries = readdirSync(fullPath, { withFileTypes: true });
      collected.push(...walkJsonFiles(fullPath, childEntries));
      continue;
    }
    if (entry.isFile() && entry.name.endsWith(".json")) {
      collected.push(fullPath);
    }
  }
  return collected;
}
function collectAbis(options) {
  const dojoRoot = process.env.DOJO_ROOT || process.cwd();
  const dojoEnv = process.env.DOJO_ENV || "dev";
  const manifestPath = join(dojoRoot, `manifest_${dojoEnv}.json`);
  const targetDir = join(dojoRoot, "target", dojoEnv);
  const allAbis = [];
  let manifest = null;
  if (!existsSync(manifestPath)) {
    console.error(`Manifest file not found: ${manifestPath}`);
    process.exit(1);
  }
  try {
    const manifestContent = readFileSync(manifestPath, "utf-8");
    manifest = JSON.parse(manifestContent);
    if (manifest.world?.abi) {
      allAbis.push(...manifest.world.abi);
    }
    if (manifest.contracts) {
      for (const contract of manifest.contracts) {
        if (contract.abi) {
          allAbis.push(...contract.abi);
        }
      }
    }
  } catch (error) {
    console.error(`Error reading manifest file: ${error}`);
    process.exit(1);
  }
  if (!existsSync(targetDir)) {
    console.warn(`Target directory not found: ${targetDir}`);
  } else {
    try {
      const dirEntries = readdirSync(targetDir, { withFileTypes: true });
      const files = walkJsonFiles(targetDir, dirEntries);
      for (const filePath of files) {
        try {
          const fileContent = readFileSync(filePath, "utf-8");
          const targetFile = JSON.parse(fileContent);
          if (targetFile.abi) {
            allAbis.push(...targetFile.abi);
          }
        } catch (error) {
          console.error(`Error reading file ${filePath}: ${error}`);
        }
      }
    } catch (error) {
      console.error(`Error reading target directory: ${error}`);
    }
  }
  const dedupedAbis = /* @__PURE__ */ new Map();
  const duplicateCounts = {};
  for (const entry of allAbis) {
    const type = typeof entry.type === "string" ? entry.type : "unknown";
    const name = typeof entry.name === "string" ? entry.name : "";
    const interfaceName = typeof entry.interface_name === "string" ? entry.interface_name : "";
    const key = `${type}::${name}::${interfaceName}`;
    if (dedupedAbis.has(key)) {
      duplicateCounts[key] = (duplicateCounts[key] ?? 1) + 1;
      continue;
    }
    dedupedAbis.set(key, entry);
  }
  const mergedAbis = Array.from(dedupedAbis.entries()).sort((a, b) => a[0].localeCompare(b[0])).map(([, value]) => value);
  if (Object.keys(duplicateCounts).length > 0) {
    console.warn("!  Duplicate ABI entries detected and ignored:");
    for (const [key, count] of Object.entries(duplicateCounts)) {
      console.warn(`   \u2022 ${key} (${count} occurrences)`);
    }
  }
  const output = {
    abi: mergedAbis,
    manifest: manifest && {
      world: manifest.world,
      base: manifest.base,
      contracts: manifest.contracts ?? [],
      models: manifest.models ?? []
    }
  };
  const paths = resolveOutputPaths(options.outputPath);
  ensureDirectory(paths.json);
  writeFileSync(paths.json, JSON.stringify(output, null, 2));
  console.log(`\u2705 ABI compilation complete!`);
  console.log(`\u{1F4C4} Output written to: ${paths.json}`);
  console.log(`\u{1F4CA} Total ABI entries: ${mergedAbis.length}`);
  const typeStats = mergedAbis.reduce((acc, item) => {
    const key = typeof item.type === "string" ? item.type : "unknown";
    acc[key] = (acc[key] ?? 0) + 1;
    return acc;
  }, {});
  for (const [abiType, count] of Object.entries(typeStats)) {
    console.log(`   \u2022 ${abiType}: ${count}`);
  }
  if (options.generateTypes) {
    generateAbiTypes(paths);
  }
}
function parseArgs(argv) {
  let generateTypes = false;
  let outputPath;
  let index = 0;
  while (index < argv.length) {
    const arg = argv[index];
    if (arg === "--generate-types") {
      generateTypes = true;
      index += 1;
      continue;
    }
    if (arg === "--output") {
      const value = argv[index + 1];
      if (!value || value.startsWith("--")) {
        console.error("Missing value for --output option");
        process.exit(1);
      }
      outputPath = value;
      index += 2;
      continue;
    }
    if (arg.startsWith("--output=")) {
      const value = arg.slice("--output=".length);
      if (!value) {
        console.error("Missing value for --output option");
        process.exit(1);
      }
      outputPath = value;
      index += 1;
      continue;
    }
    console.warn(`!  Unknown argument ignored: ${arg}`);
    index += 1;
  }
  return {
    generateTypes,
    outputPath
  };
}
var __filename = resolve(fileURLToPath(import.meta.url));
var entryPoint = process.argv[1] ? resolve(process.argv[1]) : void 0;
var isDirectExecution = entryPoint === __filename;
if (isDirectExecution) {
  const options = parseArgs(process.argv.slice(2));
  try {
    collectAbis(options);
  } catch (error) {
    console.error(`Unexpected error: ${error}`);
    process.exit(1);
  }
}
export {
  resolveOutputPaths
};
