// metagame-sdk@0.1.22/dist/esm/src/subscriptions/stores/miniGamesStore.js
import { create } from 'zustand';
import { logger } from '../../shared/utils/logger';
export const useMiniGamesStore = create((set, get) => ({
    // Initial state
    miniGames: {},
    isInitialized: false,
    lastUpdated: 0,
    totalEntities: 0,
    // Actions
    initializeStore: (entities) => {
        logger.debug('miniGamesStore: initializeStore called with', entities.length, 'entities');
        const miniGames = {};
        entities.forEach((entity) => {
            if (entity.GameMetadataUpdate) {
                const gameId = String(entity.GameMetadataUpdate.id);
                const processedGame = {
                    game_id: Number(entity.GameMetadataUpdate.id) || 0,
                    contract_address: entity.GameMetadataUpdate.contract_address || '',
                    name: entity.GameMetadataUpdate.name || '',
                    description: entity.GameMetadataUpdate.description || '',
                    developer: entity.GameMetadataUpdate.developer || '',
                    publisher: entity.GameMetadataUpdate.publisher || '',
                    genre: entity.GameMetadataUpdate.genre || '',
                    image: entity.GameMetadataUpdate.image || '',
                    color: entity.GameMetadataUpdate.color,
                    client_url: entity.GameMetadataUpdate.client_url,
                    renderer_address: entity.GameMetadataUpdate.renderer_address,
                };
                miniGames[gameId] = processedGame;
            }
        });
        logger.debug('miniGamesStore: built', Object.keys(miniGames).length, 'mini games');
        logger.debug('miniGamesStore: sample mini games:', Object.values(miniGames).slice(0, 2));
        set({
            miniGames,
            isInitialized: true,
            lastUpdated: Date.now(),
            totalEntities: entities.length,
        });
    },
    updateEntity: (entity) => {
        if (!entity.GameMetadataUpdate)
            return;
        const state = get();
        logger.debug(entity);
        const gameId = entity.GameMetadataUpdate.id.toString();
        logger.debug(entity);
        const updatedMiniGames = {
            ...state.miniGames,
            [gameId]: {
                game_id: Number(entity.GameMetadataUpdate.id),
                contract_address: entity.GameMetadataUpdate.contract_address,
                name: entity.GameMetadataUpdate.name,
                description: entity.GameMetadataUpdate.description || '', // description is already a string
                developer: entity.GameMetadataUpdate.developer,
                publisher: entity.GameMetadataUpdate.publisher,
                genre: entity.GameMetadataUpdate.genre,
                image: entity.GameMetadataUpdate.image || '',
                color: entity.GameMetadataUpdate.color,
                client_url: entity.GameMetadataUpdate.client_url,
                renderer_address: entity.GameMetadataUpdate.renderer_address,
            },
        };
        set({
            miniGames: updatedMiniGames,
            lastUpdated: Date.now(),
        });
    },
    clearStore: () => {
        set({
            miniGames: {},
            isInitialized: false,
            lastUpdated: 0,
            totalEntities: 0,
        });
    },
    updateMiniGames: (entities) => {
        const miniGames = entities.reduce((acc, entity) => {
            if (entity.GameMetadataUpdate) {
                const gameId = String(entity.GameMetadataUpdate.id);
                const processedGame = {
                    game_id: Number(entity.GameMetadataUpdate.id) || 0,
                    contract_address: entity.GameMetadataUpdate.contract_address || '',
                    name: entity.GameMetadataUpdate.name || '',
                    description: entity.GameMetadataUpdate.description || '',
                    developer: entity.GameMetadataUpdate.developer || '',
                    publisher: entity.GameMetadataUpdate.publisher || '',
                    genre: entity.GameMetadataUpdate.genre || '',
                    image: entity.GameMetadataUpdate.image || '',
                    color: entity.GameMetadataUpdate.color,
                    client_url: entity.GameMetadataUpdate.client_url,
                    renderer_address: entity.GameMetadataUpdate.renderer_address,
                };
                acc[gameId] = processedGame;
            }
            return acc;
        }, {});
        set((state) => ({
            miniGames,
            lastUpdated: Date.now(),
        }));
    },
    // Getters
    getMiniGamesByFilter: (filter) => {
        const state = get();
        let filtered = { ...state.miniGames };
        if (filter.gameAddresses && filter.gameAddresses.length > 0) {
            filtered = Object.fromEntries(Object.entries(filtered).filter(([_, game]) => filter.gameAddresses.includes(game.contract_address)));
        }
        return filtered;
    },
    getMiniGameData: (game_id) => {
        const state = get();
        const gameIdStr = game_id.toString();
        const game = state.miniGames[gameIdStr];
        return game || null;
    },
    getMiniGameByContractAddress: (contract_address) => {
        const state = get();
        const game = Object.values(state.miniGames).find((game) => game.contract_address === contract_address);
        return game || null;
    },
}));
