// metagame-sdk@0.1.22/dist/esm/src/subscriptions/stores/settingsStore.js
import { create } from 'zustand';
import { useMiniGamesStore } from './miniGamesStore';
import { logger } from '../../shared/utils/logger';
// Helper function to parse and normalize settings data
const parseSettingsData = (rawData) => {
    if (!rawData) {
        logger.debug('parseSettingsData: No raw data provided');
        return { name: '', description: '', data: {} };
    }
    try {
        // If it's already an object, use it directly
        const parsed = typeof rawData === 'string' ? JSON.parse(rawData) : rawData;
        logger.debug('parseSettingsData: Parsed data:', parsed);
        // Handle the current structure: {Name, Description, Settings: {}}
        const name = parsed.Name || '';
        const description = parsed.Description || '';
        const settings = parsed.Settings || parsed; // Fallback to the whole object if no Settings key
        const result = {
            name,
            description,
            data: settings,
        };
        logger.debug('parseSettingsData: Result:', result);
        return result;
    }
    catch (error) {
        logger.warn('Failed to parse settings data:', error, 'Raw data:', rawData);
        return { name: '', description: '', data: rawData };
    }
};
function buildSettingsFromEntities(entities) {
    const settings = {};
    // Get the mini games store to get complete game metadata
    const miniGamesStore = useMiniGamesStore.getState();
    entities.forEach((entity) => {
        if (entity.SettingsCreated?.settings_id !== undefined) {
            const settingsId = entity.SettingsCreated.settings_id.toString();
            const parsedData = parseSettingsData(entity.SettingsCreated.settings_data || entity.SettingsCreated.data);
            // First try to get game_id directly, then try to find by game_address
            let gameId = 0;
            let gameMetadata = null;
            if (entity.SettingsCreated.game_address) {
                // Try to find game by contract address
                gameMetadata = miniGamesStore.getMiniGameByContractAddress(entity.SettingsCreated.game_address);
                if (gameMetadata) {
                    gameId = gameMetadata.game_id;
                }
            }
            settings[settingsId] = {
                game_id: gameId,
                gameMetadata: gameMetadata,
                name: parsedData.name,
                description: parsedData.description,
                data: parsedData.data,
            };
        }
    });
    logger.debug('Built settings lookup from', entities.length, 'entities:', settings);
    return settings;
}
export const useSettingsStore = create((set, get) => ({
    // Initial state
    settings: {},
    isInitialized: false,
    lastUpdated: 0,
    totalEntities: 0,
    // Actions
    initializeStore: (entities) => {
        const settings = buildSettingsFromEntities(entities);
        set({
            settings,
            isInitialized: true,
            lastUpdated: Date.now(),
            totalEntities: entities.length,
        });
    },
    updateEntity: (entity) => {
        if (entity.SettingsCreated?.settings_id === undefined)
            return;
        const settingsId = entity.SettingsCreated.settings_id.toString();
        const parsedData = parseSettingsData(entity.SettingsCreated.settings_data || entity.SettingsCreated.data);
        const gameId = Number(entity.SettingsCreated.game_id) || 0;
        // Get the mini games store to get complete game metadata
        const miniGamesStore = useMiniGamesStore.getState();
        const gameMetadata = miniGamesStore.getMiniGameData(gameId);
        const settingsData = {
            game_id: gameId,
            gameMetadata: gameMetadata,
            name: parsedData.name,
            description: parsedData.description,
            data: parsedData.data,
        };
        set((state) => ({
            settings: {
                ...state.settings,
                [settingsId]: settingsData,
            },
            lastUpdated: Date.now(),
        }));
    },
    clearStore: () => {
        set({
            settings: {},
            isInitialized: false,
            lastUpdated: 0,
            totalEntities: 0,
        });
    },
    // Getters
    getSettingsByFilter: (filter) => {
        const state = get();
        let filtered = { ...state.settings };
        if (filter.settingsIds && filter.settingsIds.length > 0) {
            filtered = Object.fromEntries(Object.entries(filtered).filter(([settingsId]) => filter.settingsIds.includes(Number(settingsId))));
        }
        if (filter.gameAddresses && filter.gameAddresses.length > 0) {
            filtered = Object.fromEntries(Object.entries(filtered).filter(([_, settings]) => settings.gameMetadata &&
                filter.gameAddresses.includes(settings.gameMetadata.contract_address)));
        }
        return filtered;
    },
    getSettingsData: (settings_id) => {
        const state = get();
        const settings = state.settings[settings_id];
        return settings ? settings.data : null;
    },
}));
