// metagame-sdk@0.1.22/dist/esm/src/subscriptions/hooks/useSubscribeSettings.js
import { getMetagameClientSafe } from '../../shared/singleton';
import { useEventSubscription } from '../../shared/dojo/hooks/useEventSubscription';
import { useEffect, useMemo, useState, useCallback } from 'react';
import { useSettingsStore } from '../stores/settingsStore';
import { settingsQuery } from '../queries/sdk';
import { useEnsureMiniGamesStore } from '../utils/ensureMiniGamesStore';
import { logger } from '../../shared/utils/logger';
export function useSubscribeSettings(params = {}) {
    const client = getMetagameClientSafe();
    const { enabled = true, logging = false, gameAddresses, settingsIds, pagination } = params;
    // Ensure mini games store is initialized for gameMetadata relationships
    useEnsureMiniGamesStore();
    // Pagination state
    const pageSize = pagination?.pageSize ?? 20;
    const [currentPage, setCurrentPage] = useState(pagination?.initialPage ?? 0);
    // Sorting parameters
    const sortBy = pagination?.sortBy ?? 'game_id';
    const sortOrder = pagination?.sortOrder ?? 'asc';
    const query = useMemo(() => {
        if (!client)
            return null;
        return settingsQuery({ namespace: client.getNamespace() });
    }, [client]);
    const { entities, isSubscribed, error } = useEventSubscription(client, {
        query: query || { keys: [], entityModels: [], eventModels: [] },
        namespace: client?.getNamespace() || '',
        enabled: enabled && !!client,
        logging,
        transform: (entity) => {
            if (!client)
                return entity;
            const { entityId, models } = entity;
            const transformed = {
                entityId,
                ...models[client.getNamespace()],
            };
            // Call our store's updateEntity for real-time updates
            updateEntity(transformed);
            return transformed;
        },
    });
    logger.debug('settings entities', entities);
    const { initializeStore, updateEntity, settings, isInitialized, getSettingsByFilter, getSettingsData, } = useSettingsStore();
    // Initialize store with all entities on first load
    useEffect(() => {
        if (entities && entities.length > 0) {
            logger.debug('Initializing settings store with', entities.length, 'entities');
            initializeStore(entities);
        }
    }, [entities, initializeStore]);
    // Apply filters to get filtered settings (same logic as useMergedSettings)
    const filteredSettings = getSettingsByFilter({
        gameAddresses,
        settingsIds,
    });
    // Convert to array and sort for pagination
    const sortedSettingsArray = useMemo(() => {
        const settingsArray = Object.entries(filteredSettings);
        // Sort settings based on sortBy and sortOrder
        return settingsArray.sort(([keyA, settingA], [keyB, settingB]) => {
            let aValue;
            let bValue;
            switch (sortBy) {
                case 'game_id':
                    aValue = settingA.game_id ?? 0;
                    bValue = settingB.game_id ?? 0;
                    break;
                case 'name':
                    aValue = (settingA.name || '').toLowerCase();
                    bValue = (settingB.name || '').toLowerCase();
                    break;
                case 'description':
                    aValue = (settingA.description || '').toLowerCase();
                    bValue = (settingB.description || '').toLowerCase();
                    break;
                default:
                    return 0;
            }
            if (aValue < bValue) {
                return sortOrder === 'asc' ? -1 : 1;
            }
            if (aValue > bValue) {
                return sortOrder === 'asc' ? 1 : -1;
            }
            return 0;
        });
    }, [filteredSettings, sortBy, sortOrder]);
    // Calculate pagination values
    const totalItems = sortedSettingsArray.length;
    const totalPages = Math.ceil(totalItems / pageSize);
    const hasNextPage = currentPage < totalPages - 1;
    const hasPreviousPage = currentPage > 0;
    // Auto-adjust current page if it's beyond available pages
    useEffect(() => {
        if (totalPages > 0 && currentPage >= totalPages) {
            setCurrentPage(Math.max(0, totalPages - 1));
        }
    }, [totalPages, currentPage]);
    // Apply pagination to sorted settings
    const paginatedSettings = useMemo(() => {
        const startIndex = currentPage * pageSize;
        const endIndex = startIndex + pageSize;
        const paginatedArray = sortedSettingsArray.slice(startIndex, endIndex);
        // Convert back to record
        return Object.fromEntries(paginatedArray);
    }, [sortedSettingsArray, currentPage, pageSize]);
    // Pagination control functions
    const goToPage = useCallback((page) => {
        const clampedPage = Math.max(0, Math.min(page, totalPages - 1));
        setCurrentPage(clampedPage);
    }, [totalPages]);
    const nextPage = useCallback(() => {
        if (hasNextPage) {
            setCurrentPage((prev) => prev + 1);
        }
    }, [hasNextPage]);
    const previousPage = useCallback(() => {
        if (hasPreviousPage) {
            setCurrentPage((prev) => prev - 1);
        }
    }, [hasPreviousPage]);
    const firstPage = useCallback(() => {
        setCurrentPage(0);
    }, []);
    const lastPage = useCallback(() => {
        setCurrentPage(Math.max(0, totalPages - 1));
    }, [totalPages]);
    // Pagination controls object
    const paginationControls = useMemo(() => ({
        currentPage,
        pageSize,
        totalItems,
        totalPages,
        hasNextPage,
        hasPreviousPage,
        goToPage,
        nextPage,
        previousPage,
        firstPage,
        lastPage,
    }), [
        currentPage,
        pageSize,
        totalItems,
        totalPages,
        hasNextPage,
        hasPreviousPage,
        goToPage,
        nextPage,
        previousPage,
        firstPage,
        lastPage,
    ]);
    // Return empty state if client is not ready
    if (!client) {
        return {
            // Subscription status
            isSubscribed: false,
            error: null,
            // Store data (empty)
            settings: {},
            allSettings: {},
            getSettingsData,
            isInitialized: false,
            // Pagination controls (default state)
            pagination: {
                currentPage: 0,
                pageSize,
                totalItems: 0,
                totalPages: 0,
                hasNextPage: false,
                hasPreviousPage: false,
                goToPage,
                nextPage,
                previousPage,
                firstPage,
                lastPage,
            },
        };
    }
    return {
        // Subscription status
        isSubscribed,
        error,
        // Store data (paginated and unpaginated)
        settings: paginatedSettings, // Paginated settings for current page
        allSettings: filteredSettings, // All filtered settings (unpaginated)
        getSettingsData,
        isInitialized,
        // Pagination controls
        pagination: paginationControls,
    };
}
