// metagame-sdk@0.1.22/dist/esm/src/shared/generated/models.gen.js
export function getSchemaWithnamespace(namespace) {
    return {
        [namespace]: schemaTemplate,
    };
}
export const schemaTemplate = {
    GameCounter: {
        key: 0,
        count: 0,
    },
    GameCounterValue: {
        count: 0,
    },
    GameMetadata: {
        id: 0,
        creator_token_id: 0,
        contract_address: "",
        name: 0,
        description: "",
        developer: 0,
        publisher: 0,
        genre: 0,
        image: "",
        color: "",
    },
    GameMetadataValue: {
        creator_token_id: 0,
        contract_address: "",
        name: 0,
        description: "",
        developer: 0,
        publisher: 0,
        genre: 0,
        image: "",
        color: "",
    },
    GameRegistry: {
        contract_address: "",
        id: 0,
    },
    GameRegistryId: {
        id: 0,
        contract_address: "",
    },
    GameRegistryIdValue: {
        contract_address: "",
    },
    GameRegistryValue: {
        id: 0,
    },
    MinterCounter: {
        key: 0,
        count: 0,
    },
    MinterCounterValue: {
        count: 0,
    },
    MinterRegistry: {
        contract_address: "",
        id: 0,
    },
    MinterRegistryId: {
        id: 0,
        contract_address: "",
    },
    MinterRegistryIdValue: {
        contract_address: "",
    },
    MinterRegistryValue: {
        id: 0,
    },
    TokenClientUrl: {
        id: 0,
        client_url: "",
    },
    TokenClientUrlValue: {
        client_url: "",
    },
    TokenCounter: {
        key: 0,
        count: 0,
    },
    TokenCounterValue: {
        count: 0,
    },
    TokenMetadata: {
        id: 0,
        game_id: 0,
        minted_at: 0,
        settings_id: 0,
        lifecycle: { start: 0, end: 0 },
        minted_by: 0,
        soulbound: false,
        game_over: false,
        completed_all_objectives: false,
        has_context: false,
        objectives_count: 0,
    },
    TokenMetadataValue: {
        game_id: 0,
        minted_at: 0,
        settings_id: 0,
        lifecycle: { start: 0, end: 0 },
        minted_by: 0,
        soulbound: false,
        game_over: false,
        completed_all_objectives: false,
        has_context: false,
        objectives_count: 0,
    },
    TokenObjective: {
        id: 0,
        index: 0,
        objective_id: 0,
        completed: false,
    },
    TokenObjectiveValue: {
        objective_id: 0,
        completed: false,
    },
    TokenPlayerName: {
        id: 0,
        player_name: 0,
    },
    TokenPlayerNameValue: {
        player_name: 0,
    },
    TokenRenderer: {
        id: 0,
        renderer_address: "",
    },
    TokenRendererValue: {
        renderer_address: "",
    },
    Lifecycle: {
        start: 0,
        end: 0,
    },
};
export function getModelsMapping(namespace) {
    return {
        GameCounter: `${namespace}-GameCounter`,
        GameCounterValue: `${namespace}-GameCounterValue`,
        GameMetadata: `${namespace}-GameMetadata`,
        GameMetadataValue: `${namespace}-GameMetadataValue`,
        GameRegistry: `${namespace}-GameRegistry`,
        GameRegistryId: `${namespace}-GameRegistryId`,
        GameRegistryIdValue: `${namespace}-GameRegistryIdValue`,
        GameRegistryValue: `${namespace}-GameRegistryValue`,
        MinterCounter: `${namespace}-MinterCounter`,
        MinterCounterValue: `${namespace}-MinterCounterValue`,
        MinterRegistry: `${namespace}-MinterRegistry`,
        MinterRegistryId: `${namespace}-MinterRegistryId`,
        MinterRegistryIdValue: `${namespace}-MinterRegistryIdValue`,
        MinterRegistryValue: `${namespace}-MinterRegistryValue`,
        TokenClientUrl: `${namespace}-TokenClientUrl`,
        TokenClientUrlValue: `${namespace}-TokenClientUrlValue`,
        TokenCounter: `${namespace}-TokenCounter`,
        TokenCounterValue: `${namespace}-TokenCounterValue`,
        TokenMetadata: `${namespace}-TokenMetadata`,
        TokenMetadataValue: `${namespace}-TokenMetadataValue`,
        TokenObjective: `${namespace}-TokenObjective`,
        TokenObjectiveValue: `${namespace}-TokenObjectiveValue`,
        TokenPlayerName: `${namespace}-TokenPlayerName`,
        TokenPlayerNameValue: `${namespace}-TokenPlayerNameValue`,
        TokenRenderer: `${namespace}-TokenRenderer`,
        TokenRendererValue: `${namespace}-TokenRendererValue`,
        Lifecycle: `${namespace}-Lifecycle`,
        Context: `${namespace}-Context`,
        ContextValue: `${namespace}-ContextValue`,
        Score: `${namespace}-Score`,
        ScoreObjective: `${namespace}-ScoreObjective`,
        ScoreObjectiveCount: `${namespace}-ScoreObjectiveCount`,
        ScoreObjectiveCountValue: `${namespace}-ScoreObjectiveCountValue`,
        ScoreObjectiveValue: `${namespace}-ScoreObjectiveValue`,
        ScoreValue: `${namespace}-ScoreValue`,
        Settings: `${namespace}-Settings`,
        SettingsCounter: `${namespace}-SettingsCounter`,
        SettingsCounterValue: `${namespace}-SettingsCounterValue`,
        SettingsDetails: `${namespace}-SettingsDetails`,
        SettingsDetailsValue: `${namespace}-SettingsDetailsValue`,
        SettingsValue: `${namespace}-SettingsValue`,
        ObjectiveData: `${namespace}-ObjectiveData`,
        ObjectiveDataValue: `${namespace}-ObjectiveDataValue`,
        Owners: `${namespace}-Owners`,
        OwnersValue: `${namespace}-OwnersValue`,
        ScoreUpdate: `${namespace}-ScoreUpdate`,
        ScoreUpdateValue: `${namespace}-ScoreUpdateValue`,
        SettingsData: `${namespace}-SettingsData`,
        SettingsDataValue: `${namespace}-SettingsDataValue`,
    };
}
