// metagame-sdk@0.1.22/dist/esm/src/shared/singleton.js
import { MetagameClient } from './client';
import { clearAllStores } from '../subscriptions/stores';
import { logger } from './utils/logger';
// Store the singleton instance
let metagameClientInstance = null;
let isInitialized = false;
let currentConfig = null;
let isInitializing = false;
/**
 * Initialize the Metagame SDK with configuration
 * This should be called once at the start of your application
 */
export async function initMetagame(config) {
    // Check if config has changed (network switch)
    if (metagameClientInstance && isInitialized && currentConfig) {
        const configChanged = currentConfig.toriiUrl !== config.toriiUrl ||
            currentConfig.worldAddress !== config.worldAddress;
        if (configChanged) {
            logger.info('[initMetagame] Config changed, reinitializing SDK');
            logger.info('[initMetagame] Old config:', { toriiUrl: currentConfig.toriiUrl, worldAddress: currentConfig.worldAddress });
            logger.info('[initMetagame] New config:', { toriiUrl: config.toriiUrl, worldAddress: config.worldAddress });
            // Mark as initializing to prevent errors during transition
            isInitializing = true;
            // Clear all stores when config changes
            clearAllStores();
            // Reset the instance
            metagameClientInstance = null;
            isInitialized = false;
        }
        else {
            return metagameClientInstance;
        }
    }
    // Configure logger if provided
    if (config.logging !== undefined) {
        logger.configure(config.logging);
    }
    isInitializing = true;
    metagameClientInstance = await MetagameClient.create(config);
    isInitialized = true;
    isInitializing = false;
    currentConfig = config;
    return metagameClientInstance;
}
/**
 * Initialize the Metagame SDK synchronously (legacy method)
 * Use initMetagame() for better dojoSDK auto-initialization
 */
export function initMetagameSync(config) {
    // Check if config has changed (network switch)
    if (metagameClientInstance && isInitialized && currentConfig) {
        const configChanged = currentConfig.toriiUrl !== config.toriiUrl ||
            currentConfig.worldAddress !== config.worldAddress;
        if (configChanged) {
            logger.info('[initMetagameSync] Config changed, reinitializing SDK');
            logger.info('[initMetagameSync] Old config:', { toriiUrl: currentConfig.toriiUrl, worldAddress: currentConfig.worldAddress });
            logger.info('[initMetagameSync] New config:', { toriiUrl: config.toriiUrl, worldAddress: config.worldAddress });
            // Clear all stores when config changes
            clearAllStores();
            // Reset the instance
            metagameClientInstance = null;
            isInitialized = false;
        }
        else {
            return metagameClientInstance;
        }
    }
    // Configure logger if provided
    if (config.logging !== undefined) {
        logger.configure(config.logging);
    }
    metagameClientInstance = new MetagameClient(config);
    isInitialized = true;
    currentConfig = config;
    return metagameClientInstance;
}
/**
 * Get the initialized Metagame client instance
 * Throws an error if the SDK hasn't been initialized
 */
export function getMetagameClient() {
    if (!metagameClientInstance || !isInitialized) {
        throw new Error('Metagame SDK is not initialized. Call initMetagame() before using getMetagameClient().');
    }
    logger.debug('[getMetagameClient] Returning singleton instance with toriiUrl:', metagameClientInstance.getToriiUrl());
    return metagameClientInstance;
}
/**
 * Get the initialized Metagame client instance safely
 * Returns null if not initialized instead of throwing
 */
export function getMetagameClientSafe() {
    if (!metagameClientInstance || !isInitialized || isInitializing) {
        logger.debug('[getMetagameClientSafe] SDK not ready:', {
            hasInstance: !!metagameClientInstance,
            isInitialized,
            isInitializing
        });
        return null;
    }
    return metagameClientInstance;
}
/**
 * Check if the SDK is initialized and ready
 */
export function isMetagameReady() {
    return !!metagameClientInstance && isInitialized && !isInitializing;
}
/**
 * Reset the Metagame SDK instance
 * Useful for testing or when you need to reinitialize with different config
 */
export function resetMetagame() {
    logger.info('[resetMetagame] Resetting SDK instance and clearing stores');
    clearAllStores();
    metagameClientInstance = null;
    isInitialized = false;
    isInitializing = false;
    currentConfig = null;
}
